packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/RealmSyncUtils.kt

```diff
@@ -224,7 +224,13 @@ internal fun convertAppError(appError: AppError): Throwable {
                         ServiceException(msg)
                     }
                 }
-                ErrorCode.RLM_ERR_INVALID_SESSION -> CredentialsCannotBeLinkedException(msg)
+                ErrorCode.RLM_ERR_INVALID_SESSION -> {
+                    if (msg.contains("a user already exists with the specified provider")) {
```

Comment:
Hmm. This looks a bit shady, but I imagine that you also felt it when writing it, so I assume that there is no way around it 🤔